### ISSUE TITLE
Underscore in field name structure change

### ISSUE THREAD

**Issue by ChrisHSandN** [phpmyadmin/phpmyadmin#16595]
### Describe the bug

Attempting to change the structure of a field containing an `_` can incorrectly match the wrong field.

### To Reproduce

1. Create a table with fields `aid` and `_id`
```sql
CREATE TABLE `test` (
 `aid` tinyint(4) NOT NULL,
 `_id` tinyint(4) NOT NULL,
 PRIMARY KEY (`aid`)
) ENGINE=InnoDB 
```
2. Click on `Structure`
3. Click `Change` on the field `_id`
4. `aid` field is edited

### Expected behavior

`_id` field should be edited

### Screenshots

![image](https://user-images.githubusercontent.com/13466620/105828790-74523600-5fbb-11eb-9a94-6b5adb9803fe.png)
![image](https://user-images.githubusercontent.com/13466620/105828815-7e743480-5fbb-11eb-8e18-f1b64833974b.png)

### Server configuration

- Operating system:
- Web server: CentOS 7
- Database version: MariaDB 10.4.11
- PHP version: 7.3.14
- phpMyAdmin version: 5.0.1

### Client configuration

- Browser: Chrome Latest
- Operating system: Windows 10

**Comment by williamdes** [comment]
Thank you @ChrisHSandN for the report !
I also reproduced this issue on 5.1.0-dev.

**Comment by williamdes** [comment]
I found the issue :tada: 
And you found the perfect test case :+1: 
```sql
-- says I have a unknown char at first position
SHOW FULL COLUMNS FROM `test` LIKE '_id'
-- says I have the `_` char at the first position
SHOW FULL COLUMNS FROM `test` LIKE '\_id'
```

To reproduce, have a column named what you want, then create another one containing a `_` at least once at start.
Then be sure to have the column having a  `_` last.

**Comment by williamdes** [comment]
I added tests in 0a1d09b282ecc0f36940441e313b1e02707c90b4 and fixed it in 8917e77bba60209b8e767312b7427a23df77614b
Next 5.1.0 release will contain this fix.
